### PR TITLE
Return Databricks Query History API id as part of AdapterResponse

### LINKED ISSUE #279: Return query id as part of adaptor response for dbt
### Describe the feature
Currently the query_id column in the run_results.json is only used by Snowflake and is null for Databricks queries. Please include the query_id of the Databricks query so that we can effectively match a dbt run result to its matching query from the Databricks query history API.

### Describe alternatives you've considered
String matching the query text but its hugely inefficient. 

### Who will this benefit?
Any dbt-databricks users doing query analysis 

### Are you interested in contributing this feature?
Not currently.

### PR DESCRIPTION
**Pull request by susodapop**
Resolves #279

### Description

This pull request populates the `query_id` field of `AdapterResponse` with the Query History API key. 

### Checklist

- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-databricks next" section.

### CODE REVIEW

**Comment by susodapop** [comment]
Integration test check fails right now because of an unrelated model contract test.

**Comment by susodapop** [comment]
The model contract test failed because dbt-core has updated their warning message from "Constraint validation failed" to "Contract enforcement failed" and we are checking for exact exception language:

https://github.com/dbt-labs/dbt-core/issues/7335

**Comment by susodapop** [comment]
https://github.com/databricks/dbt-databricks/pull/378 fixed the failing integration test. I rebased these changes off `main` to capture that fix.

**Review by andrefurlan-db — APPROVED** [review]
👍

File: dbt/adapters/databricks/connections.py [hunk 1]
```diff
@@ -726,6 +743,12 @@ def exponential_backoff(attempt: int) -> int:
             retry_timeout=(timeout if timeout is not None else exponential_backoff),
         )
 
+    @classmethod
+    def get_response(cls, cursor: DatabricksSQLCursorWrapper) -> DatabricksAdapterResponse:
```
Comment (benc-db) [inline dbt/adapters/databricks/connections.py:747]: Where is this used?  I don't see it in this PR.

Comment (susodapop) [reply dbt/adapters/databricks/connections.py:747]: `get_response` is called by the `add_query` and `execute` methods in this same file. Previously we inherited the `get_response` behaviour of databricks-spark (which we are still a fork of...for now).

Comment (benc-db) [reply dbt/adapters/databricks/connections.py:747]: Ack.  Github PRs plus python not using an override keyword makes such uses hard to spot, thanks :).

**Review by benc-db — APPROVED** [review]

**Comment by susodapop** [comment]
I synced the code with `main` to fix merge conflicts. Still waiting for our integration tests to pass (flaky DECO infra) before merging.

**Comment by susodapop** [comment]
Integration tests failed because python models don't have a `hex_query_id`. I'm updating the code.

**Pull request merged**